mirar/processors/photcal/zp_calculator/color_term_zp_calculator.py

```diff
@@ -31,6 +32,61 @@ def line_func(theta, x):
     return (slope * x) + intercept
 
 
+def line_func_curve_fit(data: np.ndarray, slope: float, intercept: float) -> np.ndarray:
+    """linear model to hand to scipy curve_fit"""
+    return slope * data + intercept
+
+
+def solve_odr(
+    x: np.ndarray,
+    y: np.ndarray,
+    x_err: np.ndarray,
+    y_err: np.ndarray,
+    firstguess_color_zp: tuple,
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Solve for the zero point and color term using scipy.odr.
+    :param x: color
+    :param y: ref_mag - img_mag
+    :param x_err: uncertainty in color
+    :param y_err: uncertainty in ref_mag - img_mag
+    :param firstguess_color_zp: first guess at the color and zero-point values
+    :return: best fit color and zero-point values, and their uncertainties
+    """
+    # setup: remove sources with 0 uncertainty (or else scipy.odr won't work)
+    zero_mask = (y_err == 0) | (x_err == 0)
+    if np.sum(zero_mask) != 0:
+        logger.debug(
+            f"Found {np.sum(zero_mask)} source(s) with zero reported "
+            f"uncertainty, removing them from calibrations."
+        )
+        x, y = x[~zero_mask], y[~zero_mask]
+        x_err, y_err = x_err[~zero_mask], y_err[~zero_mask]
+
+    line_model = Model(line_func_odr)
+    data = RealData(x, y, sx=x_err, sy=y_err)
+    odr = ODR(data, line_model, beta0=firstguess_color_zp)
+    out = odr.run()
+    return out.beta, out.sd_beta
+
+
+def solve_curve_fit(
+    x: np.ndarray, y: np.ndarray, _, y_err: np.ndarray, firstguess_color_zp: tuple
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Solve for the zero point and color term using scipy.curve_fit.
+    :param x: color
+    :param y: ref_mag - img_mag
+    :param y_err: uncertainty in ref_mag - img_mag
+    :param firstguess_color_zp: first guess at the color and zero-point values
```

Comment:
maybe should note that a first guess is required for ODR, but not for scipy.optimize.curve_fit

Reply:
We now require first guess for both odr and curve_fit (curve_fit itself doesnt, but mirar does. Its a good thing to provide first guesses anyway).